tools/dissector/ws_common.h

```diff
@@ -11,27 +11,110 @@
 MAJOR * 1000000 + MINOR * 1000 + MICRO
 #define WIRESHARK_VERSION WIRESHARK_VERSION_NUMBER(VERSION_MAJOR,VERSION_MINOR,VERSION_MICRO)
 
+
+#if WIRESHARK_VERSION >= WIRESHARK_VERSION_NUMBER(2, 5, 0)
+// find_conversation after 2.5/2.6 takes endpoint_type instead of port_type
+// but also provides a shortcut function which we will use instead.
+inline conversation_t *ws_find_conversation(packet_info *pinfo) {
+  return find_conversation_pinfo(pinfo, 0);
+}
+#else // if 2.4.x or before
+inline conversation_t *ws_find_conversation(packet_info *pinfo) {
+  return find_conversation(
+    pinfo->fd->num, &pinfo->src, &pinfo->dst,
+    pinfo->ptype, pinfo->srcport, pinfo->destport, 0
+  );
+}
+#endif
+
+#if WIRESHARK_VERSION >= WIRESHARK_VERSION_NUMBER(1,4,0)
+// 1.4 introduced find_or_create_conversation
+inline conversation_t *ws_find_or_create_conversation(packet_info *pinfo) {
+  return find_or_create_conversation(pinfo);
+}
+#else // if 1.3.x or before
+inline conversation_t *ws_find_or_create_conversation(packet_info *pinfo) {
+  conversation_t *conv = NULL;
+  if (!(conv = ws_find_conversation(pinfo))) {
+    conv = conversation_new(
+      pinfo->num, &pinfo->src, &pinfo->dst,
+      pinfo->ptype, pinfo->srcport, pinfo->destport, 0
+    );
+  }
+  return conv;
+}
+#endif
+
+#if WIRESHARK_VERSION >= WIRESHARK_VERSION_NUMBER(2, 2, 0)
+inline guint8* ws_tvb_get_string(wmem_allocator_t* alloc, tvbuff_t *tvb,
+                                 gint offset, gint length)
+{
+  return tvb_get_string_enc(alloc, tvb, offset, length, ENC_ASCII);
+}
+#else
+# define ws_tvb_get_string ::tvb_get_string
+#endif
+
+#if WIRESHARK_VERSION >= WIRESHARK_VERSION_NUMBER(2, 2, 0)
+# define WS_DISSECTOR_RETURN_TYPE int
+# define WS_DISSECTOR_RETURN_VALUE return 0; //TODO
```

Comment:
What needs to be done to resolve the //TODO comment?

Reply:
That was there before me, I tried to figure out early on because it was a TODO but moved onto the sample dissection because I could understand that better. It needs to return the number of bytes it dissected. There is a related marco for the tcp dissection that returns how many more bytes the dissector needs. I don't know how applicable these are to this dissector though, given that it has worked as is, but I will look into it this week.